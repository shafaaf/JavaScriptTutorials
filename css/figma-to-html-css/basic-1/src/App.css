:root {
    --clr-primary: #12e7c0;
    --clr-graylight: #f3f3f3;
    --clr-graymed: #c4c4c4;
    --border-radius: 0.5rem;
}

/* To apply universally */
*,
*:before,
*:after {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
}

.body {
    background: var(--clr-graylight);
}

/* Have some breathing room on the main content from edges */
main {
    margin: 2rem;
}

h1 {
    text-align: center;
}

/* Will hold all the cards */
.wrapper {
    counter-reset: number;
    display: flex;
    /* widths of the cards remain fixed, wont change on less screen size, if less space, new line */
    /* Imp: really good to achieve responsiveness */
    flex-wrap: wrap;
    justify-content: center; /* All cards will be at the center */
    margin: 4rem 0;
    /*row-gap: 2rem;*/
}

/* each card */
.container {
    box-shadow: 0px 0px 20px var(--clr-graymed);
    border-radius: var(--border-radius);
    padding: 2rem;
    width: 300px; /* Dont want these to stretch as screen grows and so fixed */
    margin: 0 0.5rem 4rem; /* Breathing room between reach of them */
    background: white;
    text-align: center;
    /* want number circles which appear with position absolute above this,
    and so will need to be in reference to this container */
    position: relative;
}

/* Green numbered circle */
.container::before {
    counter-increment: number; /* Increment number by 1 */
    content: counter(number);
    width: 4rem;
    height: 4rem;
    background: var(--clr-primary);
    position: absolute;
    display: block;
    border-radius: 50%;
    top: -2rem;
    left: 7.4rem;
    font-size: 3rem;
    font-weight: bold;
    line-height: 4rem;
}

.title {
    margin: 2rem 0 1rem 0;
}
