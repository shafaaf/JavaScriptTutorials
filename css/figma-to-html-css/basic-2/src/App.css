
* {
    box-sizing: border-box; /* so that height and width of all elements will be just as we want */
    margin: 0;
}
