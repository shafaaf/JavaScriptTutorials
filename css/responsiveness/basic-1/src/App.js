const App = () => (
    <div>
        <img src="img/banner.png" alt=""/>
        Learn React
    </div>
);

export default App;

// Design for cellphone first
