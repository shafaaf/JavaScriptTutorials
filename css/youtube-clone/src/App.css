*, *::before, *::after {
    box-sizing: border-box;
    font-family: Roboto, Arial;
}

body {
    margin: 0;
}

/* TODO: Learn how grid works here */
.video-section {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
    gap: 3rem 1rem;
    padding: 3rem 0;
    margin: 0 1.5rem;
    border-top: 4px solid #CCC;
}

.video-section:first-child {
    border-top: none;
}

.video-container {
    display: flex;
    flex-direction: column; /* video-bottom-section stuff like channel icon, video title etc appears at bottom */
}

.thumbnail {
    position: relative; /* Unsure why relative here */
    display: flex;
}

.thumbnail::before { /* for the time thing */
    content: attr(data-duration);
    position: absolute; /* TODO: Unsure why working */
    background-color: rgba(0, 0, 0, 0.85);
    color: white;
    right: 5px;
    bottom: 5px;
    padding: .1em .3em;
    border-radius: .3em;
    font-size: .9em;
}

.thumbnail-image {
    width: 100%;
    height: 100%;
    min-width: 250px;
    min-height: 150px;
    background-color: grey; /* Show grey box which shows up when loading */
}

/*bottom of video stuff*/

.video-bottom-section {
    display: flex;
    align-items: flex-start;
    margin-top: 1rem;
}

.channel-icon {
    margin-right: 0.75rem;
    border-radius: 50%;
    width: 40px;
    height: 40px;
    background-color: grey; /* Show grey box which shows up when loading */
}

.video-details {
    display: flex;
    flex-direction: column;
}

.video-title {
    font-size: 1.1rem;
    font-weight: bolder;
    margin-bottom: 0.5rem;
    text-decoration: none;
    color: black;
}

.video-channel-name {
    margin-bottom: .1rem;
    text-decoration: none;
    transition: color 150ms;
}

.video-channel-name:hover {
    color: #111;
}

.video-metadata,
.video-channel-name {
    color: #555;
}
