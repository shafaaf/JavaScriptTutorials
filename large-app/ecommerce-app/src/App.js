import React from 'react';
import data from './data';
import Products from "./ components/Products";
import Filter from "./ components/Filter";

class App extends React.Component {
    constructor() {
        super();
        this.state = {
            products: data.products,
            // user selected
            size: "",
            sort: ""
        };
        console.log("products is: ", this.state.products);
    }

    sortProducts = e => { // sort filtered product list by size
        const sortBy = e.target.value;
        console.log("calling sortProducts: ", sortBy);
        this.setState({
            sort: sortBy,
            products: this.state.products.sort((a, b) => {
                if (sortBy  === "lowest") {
                    return a.price < b.price? -1:1;
                } else if (sortBy  === "highest") {
                    return a.price < b.price? 1:-1;
                } else if (sortBy  === "latest") {
                    return a._id < b._id? -1:1;
                }
            })
        });
    };

    filterProductsBySize = e => { // filter all products by size
        const size = e.target.value;
        console.log("calling filterProducts: ", size);

        if (size === "ALL") { // show all products
            console.log("All selected");
            this.setState({
                products: data.products
            });
        } else { // filter by some size
            this.setState({
                size: size,
                products: data.products.filter(product =>
                    product.availableSizes.indexOf(size) >= 0
                )
            });
        }
    };

    render() {
        return (
            <div className="App">
                <div className="grid-container">
                    <header>
                        <a href="/">React Shopping Cart</a>
                    </header>
                    <main>
                        <div className="content">
                            <div className="main">
                                <Filter
                                    count = {this.state.products.length}
                                    size = {this.state.size}
                                    sort = {this.state.sort}
                                    filterProductsBySize = {this.filterProductsBySize}
                                    sortProducts = {this.sortProducts}
                                />
                                <Products products = {this.state.products}/>
                            </div>
                            <div className="sidebar">
                                Cart Items
                            </div>
                        </div>
                    </main>

                    <footer>
                        All rights reserved.
                    </footer>

                </div>
            </div>
        );
    }
}

export default App;
