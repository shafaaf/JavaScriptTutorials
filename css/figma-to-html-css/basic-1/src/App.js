import  './App.css';

const App = () => {
    const titles = [
        "Pick a Template",
        "Make it your own",
        "Share it out"
    ];
    const descriptions = [
        "Choose from a selection of over 300 templates. Perfect for Birthdays, Graduation, Anniversaries and more!",
        "Customize your card by uploading images, videos and writing a personal message!",
        "Send your card out by email or text. You can send your card out immediately or schedule it for the perfect time and date."
    ];
    return (
        <main>
            <h1>Get Started</h1>
            <div className="wrapper"> {/* Will hold the cards */}
                {
                    titles.map((title, i) => (
                        <div key={i} className="container"> {/* each card */}
                            <h2 className="title">{title}</h2>
                            <p className="description">{descriptions[i]}</p>
                        </div>
                    ))
                }
            </div>
        </main>
    );
}

export default App;
