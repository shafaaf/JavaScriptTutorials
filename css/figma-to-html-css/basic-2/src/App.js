import {ReactComponent as SearchIcon} from './images/search-icon.svg';
import {ReactComponent as CloseIcon} from './images/close-icon.svg';
import styles from "./App.module.scss";
import "./App.css";

import photo from './images/photo.png';
import instagram from './images/instagram.png';
import behance from './images/behance.png';
import dribble from './images/dribble.png';

const App = () => (
    <div className={styles.fullContainer}>
        <div className={styles.searchBox}>
            <input type="text" placeholder="Search here..."/>
        </div>

        <div className={styles.navContainer}>
            <div className={styles.wrapper}>
                <div className={styles.navLogo}>D.</div>
                <ul className={styles.navUnorderedList}>
                    <li className={styles.navListItems}>
                        <a className={styles.navListItemsLinks} href="#">blog</a>
                    </li>
                    <li className={styles.navListItems}>
                        <a className={styles.navListItemsLinks} href="#">contact</a>
                    </li>
                    <li className={styles.navListItems}>
                        <p className={styles.navListItemsLinks} href="#">about</p>
                    </li>
                    <li className={styles.navListItems}>
                        <a className={styles.navListItemsLinks} href="#">
                            <SearchIcon />
                            <CloseIcon />
                        </a>
                    </li>
                </ul>
            </div>
        </div>

        <div className={styles.mainContentContainer}>
            <div className={styles.authorContent}>
                <div>
                    <h1 className={styles.authorHeader}>Hi, Iâ€™m Darlene</h1>
                    <p className={styles.authorPara}>
                        I design beautiful websites & mobile apps that modern trends
                        demand
                    </p>
                    <button className={styles.authorContact}>
                        Contact Me
                    </button>
                </div>
                <div className="hero-image">
                    <img src={photo} alt="photo"/>
                    <div className="photo-bg"></div>
                </div>
            </div>

            <div className="social-icons">
                <a href="#">
                    <img src={instagram} alt="instagram"/>
                </a>
                <a href="#">
                    <img src={behance} alt="behance"/>
                </a>
                <a href="#">
                    <img src={dribble} alt="dribble"/>
                </a>
            </div>

        </div>


    </div>
);

export default App;
