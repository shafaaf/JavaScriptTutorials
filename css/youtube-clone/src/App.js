import './App.css';

const App = () => {
    const array1 = [1, 4, 9, 16];

    // return (
    //     <div>
    //     {array1.map((value, index) => (
    //
    //     ))}
    // </div>);


    return (
        <div className="videos">
            <section className="video-section">

                <article className="video-container">
                    <a href="#" className="thumbnail" data-duration="12:24">
                        <img src="https://unsplash.it/250/150?gravity=center" alt="random video image"
                             className="thumbnail-image"/>
                    </a>

                    <div className="video-bottom-section">
                        {/*left side - channel's profile image*/}
                        <a href="#">
                            <img className="channel-icon" src="https://unsplash.it/36/36?gravity=center"
                                 alt="random user image"/>
                        </a>
                        {/*right side - video details*/}
                        <div className="video-details">
                            <a href="#" className="video-title">Video Title</a>
                            <a href="#" className="video-channel-name">Channel Name</a>
                            <div className="video-metadata">
                                <span>12K views • 1 week ago</span>
                            </div>
                        </div>
                    </div>
                </article>

                <article className="video-container">
                    <a href="#" className="thumbnail" data-duration="12:24">
                        <img className="thumbnail-image" src="https://unsplash.it/250/150?gravity=center"
                             alt="random video image"/>
                    </a>

                    <div className="video-bottom-section">
                        {/*left side - user image*/}
                        <a href="#">
                            <img className="channel-icon" src="https://unsplash.it/36/36?gravity=center"
                                 alt="random user image"/>
                        </a>
                        {/*right side - video details*/}
                        <div className="video-details">
                            <a href="#" className="video-title">Video Title</a>
                            <a href="#" className="video-channel-name">Channel Name</a>
                            <div className="video-metadata">
                                <span>12K views • 1 week ago</span>
                            </div>
                        </div>
                    </div>
                </article>
            </section>
        </div>
    )
};

export default App;
