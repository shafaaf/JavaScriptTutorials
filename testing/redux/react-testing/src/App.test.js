import React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('App Component', () => {
  let props;

  beforeEach(() => {
    props = {
      users: {
        users: [],
        loading: false,
        error: false
      }
    };
  });

  it('renders without crashing', () => {
    const tree = renderer.create(<App {...props} />)

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders an error message when a network error occurs', () => {
    props.users.error = true;
    const tree = renderer.create(<App {...props} />)

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('calls the getUsers function when the button is clicked', () => {
    props.getUsers = jest.fn();
    const wrapper = shallow(<App {...props} />);
    const spy = jest.spyOn(wrapper.instance().props, 'getUsers');

    wrapper.find('button').simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('renders the User component correctly', () => {
    props.users.users = [
      {
        id: 1,
        name: 'foo'
      }
    ]
    const wrapper = shallow(<App {...props} />);

    expect(wrapper.find('User').length).toBe(1);
  });
});
