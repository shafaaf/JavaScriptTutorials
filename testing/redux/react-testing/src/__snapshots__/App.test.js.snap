// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders an error message when a network error occurs 1`] = `
<div
  className="App"
>
  <button
    className="loadButton"
    disabled={false}
  >
    Load Users
  </button>
  <div>
    A network error occured
  </div>
</div>
`;

exports[`App Component renders without crashing 1`] = `
<div
  className="App"
>
  <button
    className="loadButton"
    disabled={false}
  >
    Load Users
  </button>
</div>
`;
